### PR TITLE
update libMLIR_h bindings and fix bindings update step n°2

### LINKED ISSUE #201: Error on regenerating "libMLIR_h.jl" on GitHub runner
The "Regenerate MLIR bindings" is currently failing when regenerating the "libMLIR_h.jl" file. Logs show that it fails on downloading the `GCCBootstrap` artifacts because it fails on creating a temporary directory in `~/.julia/artifacts/` due to the path being read-only.

https://github.com/EnzymeAD/Reactant.jl/actions/runs/11541259802/job/32123044918

```
 Downloading artifact: GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked
     Failure artifact: GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked
ERROR: LoadError: IOError: mktempdir("/home/runner/.julia/artifacts"): read-only file system (EROFS)
```

CC @Pangoraw

### PR DESCRIPTION
**Pull request by Pangoraw**
close #201 

- **fetch lazy artifacts in pre-process step**
- **update libMLIR_h.jl**

### CODE REVIEW

File: .github/workflows/regenerate-mlir-bindings.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         run: |
           import Pkg
           Pkg.add("JuliaFormatter")
-      - run: julia --project=. -e 'using Pkg; pkg"instantiate"'
+      - run: julia --project=. -e 'using Pkg; Pkg.instantiate(); using Clang; Clang.JLLEnvs.get_system_includes()'
```
Comment (mofeing) [inline .github/workflows/regenerate-mlir-bindings.yml:25]: I don't understand why this should fix it?

Comment (Pangoraw) [reply .github/workflows/regenerate-mlir-bindings.yml:25]: It downloads the lazy artifact before running inside the bazel sandbox

Comment (Pangoraw) [reply .github/workflows/regenerate-mlir-bindings.yml:25]: New error https://github.com/EnzymeAD/Reactant.jl/actions/runs/11549131512/job/32141659071

**Comment by mofeing** [comment]
@Pangoraw I think we can revert the change of this commit on "regenerate-mlir-bindings.yml" since I was getting the same error by just doing `]update` and `]precompile`.

mmm I believe the problem originates in `bazel` calling `julia` without the `--project` flag.

**Pull request merged**